ssl/statem/extensions.c

```diff
@@ -177,6 +231,9 @@ static const EXTENSION_DEFINITION ext_defs[] = {
         TLSEXT_TYPE_ec_point_formats,
         SSL_EXT_CLIENT_HELLO | SSL_EXT_TLS1_2_SERVER_HELLO
         | SSL_EXT_TLS1_2_AND_BELOW_ONLY,
+#ifndef OPENSSL_NO_ECH
+        OSSL_ECH_HANDLING_COMPRESS,
```

Comment:
You better merge this flag with the previous ones SSL_EXT_*.
It will spare a new field in this structure.

Reply:
We could do that, but, if so, I think not yet - would  prefer to get feedback on the general scheme of compile-time setting these values first. I also think `ech_handlng` and `context` differ enough that merging them may be unwise but let's see how we go.

Reply:
IMO having these always compiled in and merged with the SSL_EXT_* would be OK.

Reply:
> IMO having these always compiled in and merged with the SSL_EXT_* would be OK.

Not sure that's a good plan. Isn't the `context` field passed in to e.g. all 'tls_construct_ctos_*` and `tls_parse_stoc_*` functions, and I see lines like the one below (in `extensions_clnt.c:1737`):

```
    if (context == SSL_EXT_TLS1_3_CERTIFICATE_REQUEST) {
```
There's not many of those but I easily found 10 so those'd all need fixing if we added in the ECH handling bits to the same field.

Reply:
These code shortcuts deserve a separated cleanup PR.